core/state_transition.go

```diff
@@ -105,10 +107,10 @@ func (result *ExecutionResult) Revert() []byte {
 }
 
 // IntrinsicGas computes the 'intrinsic gas' for a message with the given data.
-func IntrinsicGas(data []byte, contractCreation, isHomestead bool, isEIP2028 bool) (uint64, error) {
+func IntrinsicGas(data []byte, accessList *types.AccessList, isContractCreation bool, isHomestead, isEIP2028 bool) (uint64, error) {
```

Comment:
Should we just pass `params.Rules` to this function instead of passing `isXXX` for each fork? I think this function will only ever get messier, might be a good time to prevent that.

Reply:
It's used from the txpool also, which doesn't have params.Rules per se

Reply:
Yes, but we can easily make it available in txpool. We can create `Rules` from `ChainConfig` and block number on every block new and use it to validate. Passing `types.Message` won't work for that though, we don't want to call `AsMessage` on every tx just for this.